### PR TITLE
add test to verify that the server does not support TLS 1.3

### PR DESCRIPTION
**Pull request by tomato42**
### Description

test to verify that the server doesn't support TLS 1.3 or has that support disabled

### Motivation and Context

script to verify that setting the maximum supported version to lower than TLS 1.3 when TLS 1.3 is implemented has effect

### Checklist

- [x] I have read the [CONTRIBUTING.md](https://github.com/tomato42/tlsfuzzer/blob/master/CONTRIBUTING.md) document and my PR follows [change requirements](https://github.com/tomato42/tlsfuzzer/blob/master/CONTRIBUTING.md#change-requirements) therein
- [x] the changes are also reflected in documentation and code comments
- [x] all new and existing tests pass (see Travis CI results)
- [ ] [test script checklist](https://github.com/tomato42/tlsfuzzer/wiki/Test-script-checklist) was followed for new scripts
- [x] new test script added to `tlslite-ng.json` and `tlslite-ng-random-subset.json`
- [ ] new and modified scripts were ran against popular TLS implementations:
  - [ ] OpenSSL
  - [x] NSS
  - [ ] GnuTLS
- [x] required version of tlslite-ng updated in requirements.txt and README.md

---
This change is [Reviewable](https://reviewable.io/reviews/tomato42/tlsfuzzer/561)

### CODE REVIEW

**Comment by tomato42** [comment]
we should advertise all known TLS 1.3 ciphers here

File: scripts/test-tls13-non-support.py [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+        node = node.add_child(FinishedGenerator())
+        node = node.add_child(ExpectChangeCipherSpec())
+        node = node.add_child(ExpectFinished())
+        node = node.add_child(ApplicationDataGenerator(
+            bytearray(b"GET / HTTP/1.0\r\n\r\n")))
+        node = node.add_child(ExpectApplicationData())
+        node = node.add_child(AlertGenerator(AlertLevel.warning,
+                                             AlertDescription.close_notify))
+        node = node.add_child(ExpectAlert())
+        conversations["try negotiating (128, {0}) - draft version"
```
Comment (Jakuje) [inline scripts/test-tls13-non-support.py]: The above code lists `(127, tls_ver)` and here is `(128, {0})`. Without TLS 1.3 knowledge, this looks suspicious and would deserve at least a comment.

File: scripts/test-tls13-non-support.py [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+from tlsfuzzer.runner import Runner
+from tlsfuzzer.messages import Connect, ClientHelloGenerator, \
+        ClientKeyExchangeGenerator, ChangeCipherSpecGenerator, \
+        FinishedGenerator, ApplicationDataGenerator, AlertGenerator
+from tlsfuzzer.expect import ExpectServerHello, ExpectCertificate, \
+        ExpectServerHelloDone, ExpectChangeCipherSpec, ExpectFinished, \
+        ExpectAlert, ExpectApplicationData, ExpectClose, \
+        ExpectEncryptedExtensions, ExpectCertificateVerify, \
+        ExpectNewSessionTicket, ExpectServerKeyExchange
```
[the remark this answers is not in the document]
Comment (tomato42) [reply scripts/test-tls13-non-support.py]: done

File: scripts/test-tls13-non-support.py [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+
+from tlslite.constants import CipherSuite, AlertLevel, AlertDescription, \
+        TLS_1_3_DRAFT, GroupName, ExtensionType, SignatureScheme
```
[the remark this answers is not in the document]
Comment (tomato42) [reply scripts/test-tls13-non-support.py]: done

File: scripts/test-tls13-non-support.py [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+from tlslite.keyexchange import ECDHKeyExchange
```
[the remark this answers is not in the document]
Comment (tomato42) [reply scripts/test-tls13-non-support.py]: done

File: scripts/test-tls13-non-support.py [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+from tlsfuzzer.utils.lists import natural_sort_keys
+from tlslite.extensions import KeyShareEntry, ClientKeyShareExtension, \
```
[the remark this answers is not in the document]
Comment (tomato42) [reply scripts/test-tls13-non-support.py]: done

File: scripts/test-tls13-non-support.py (hunk 1 above)
Comment (tomato42) [reply scripts/test-tls13-non-support.py]: done

**Comment by tomato42** [comment]
@Jakuje all should be fixed, please recheck

File: scripts/test-tls13-non-support.py (hunk 1 above)
Comment (Jakuje) [reply scripts/test-tls13-non-support.py]: This still lists 128 while the values above are 127 (as correctly described in the comment).

Comment (tomato42) [reply scripts/test-tls13-non-support.py]: I completely missed that part, thanks! should be fixed now

**Comment by tomato42** [comment]
thanks! (I forgot sig_algs_cert before)

**Pull request merged**